pkg/utils/download.go

```diff
@@ -48,33 +74,119 @@ func DownloadFromTarGzURL(URL string, destination string) error {
 	return err
 }
 
+func getURLToDownloadReleaseAsset(URL string, gitCredentials GitCredentials) (string, error) {
+	URLSlice := strings.Split(URL, "/")
+
+	domain := URLSlice[2]
```

Comment:
Should there be a check on the length of URLSlice? We use up to index 7 without verifying it has 7 elements.

Reply:
Thanks, I'll add a test case in